src/Account.sol

```diff
@@ -249,19 +271,41 @@ contract Account is IAccount, Auth, OpsReady {
                     amount := calldataload(_inputs.offset)
                 }
                 _modifyAccountMargin({_amount: amount});
-            } else {
-                // Command.ACCOUNT_WITHDRAW_ETH
+            } else if (_command == Command.ACCOUNT_WITHDRAW_ETH) {
                 uint256 amount;
                 assembly {
                     amount := calldataload(_inputs.offset)
                 }
                 _withdrawEth({_amount: amount});
+            } else if (_command == Command.UNISWAP_V3_SWAP) {
+                // Command.UNISWAP_V3_SWAP
+                uint256 amountIn;
+                uint256 amountOutMin;
+                bytes calldata path = _inputs.toBytes(2);
+                assembly {
+                    amountIn := calldataload(_inputs.offset)
+                    amountOutMin := calldataload(add(_inputs.offset, 0x20))
+                    // 0x40 offset is the path; decoded above
+                }
+                _uniswapV3Swap({
+                    _amountIn: amountIn,
+                    _amountOutMin: amountOutMin,
+                    _path: path
+                });
+            } else {
+                // Command.PERMIT2_PERMIT
+                IPermit2.PermitSingle calldata permitSingle;
+                assembly {
+                    permitSingle := _inputs.offset
+                }
+                bytes calldata data = _inputs.toBytes(6); // PermitSingle takes first 6 slots (0..5)
+                PERMIT2.permit(msg.sender, permitSingle, data);
             }
         } else {
             /// @dev only owner and delegate(s) can execute the following commands
             if (!isAuth()) revert Unauthorized();
 
-            if (commandIndex < 4) {
+            if (commandIndex < 6) {
```

Comment:
**Question:** Why add new commands earlier up the "switch statement" chain rather than later down? Is it to save gas as you try to put what is expected to be the most frequently used commands at the top? Because otherwise it seems like it adds extra complexity to the SC and off-chain changes required for an upgrade. Maybe you have a different reason?

Reply:
So there are a couple of reasons for inserting these specific commands into the list instead of just appending them.

1. These commands can only be executed by the owner (and not delegates), and previously I sectioned owner-only commands at the start of the list
>  It is possible to append these new commands and restrict execution to only the owner, but ultimately, it would've been less efficient from a contract point of view
2. The reason I sectioned the owner-only commands at the beginning of the list is that, as you observed, they are often called much more often than the others, so it is a gas optimization

Reply:
Ah okay, nice, makes sense!